### ISSUE TITLE
Polly ListAvailableVoices fails to return any voices

### ISSUE THREAD

**Issue by ambitslix** [aws/aws-sdk-cpp#1817]
- [x] I've gone though [Developer Guide](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/welcome.html) and [API reference](http://sdk.amazonaws.com/cpp/api/LATEST/index.html)
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-cpp/issues) and didn't find any solution
  
**Describe the bug**
polly_sample application fails to return any voices.

**SDK version number**
commit 3d74b3868716f58bbeddef6288c3ed21e2ffedc4 (HEAD -> main, tag: 1.9.149

**Platform/OS/Hardware/Device**
Linux/Ubuntu

**To Reproduce (observed behavior)**
build SDK, run polly_sample

**Expected behavior**
Should return the list of available voices or say why there isn't any available.

**Logs/output**
```
available devices are: 
[0] default audio output device   Driver: Linux (Pulse Audio)
please select deviceid to play output to:
0
available voices are: 
please select voice you would like me to render.
```

**Additional context**

**Comment by ambitslix** [comment]
One solution to this problem is to add permissions for the IAM user for AwsPollyFullAccess or otherwise. However polly+sample should return the response from aws api which would shed light on what the issue is. Other wise using aws-cli:

```
aws polly describe-voices
```

reveals the probem. All of this is probably obvious to some familiar with aws but for some who isn't it's just a bad experience.

**Comment by KaibaLopez** [comment]
Hi @ambitslix ,  just so to make sure I understand the issue here, the problem you are having is that the sample is basically silently failing right? As in, your expecting that it would tell you "Failed because you don't have the correct permissions" or something similar.
We would really appreciate the patch to the polly_sample actually if you have one in mind, we'll be happy to review it.

**Comment by ambitslix** [comment]
Hi @KaibaLopez,

Yes the aws api properly returns the response of the permission issue so TextToSpeechManager::ListAvailableVoices should do the same in polly_sample. I believe it's using https://docs.aws.amazon.com/polly/latest/dg/API_DescribeVoices.html call. I will provide a patch for review, I first have to investigate why the response is not displayed.

**Comment by KaibaLopez** [comment]
@ambitslix ,
Can you link me to the polly sample you are referring to?

**Comment by ambitslix** [comment]
@KaibaLopez ,

Yep, there's only one

https://github.com/aws/aws-sdk-cpp/blob/main/aws-cpp-sdk-polly-sample/main.cpp

**Comment by KaibaLopez** [comment]
Ok, my bad for the confusion, I finally got the whole picture and yea I see your problem with it... A couple of things to note here:

* You are right, the ListAvailableVoices from the text-to-speech manager is just a wrapper over polly's DescribeVoices API call. But unlike the rest of the SDK's "raw" API calls, it doesn't return a "success/failure" output, it just returns the array, meaning that it will just return an empty list if for whatever reason that call fails.
* To see the error, you'd have to generate the logs and there you'd see why the call failed. Admittedly this isn't new-user friendly, but just wanted to mention that it is there.
```
Aws::SDKOptions options;
options.loggingOptions.logLevel = Aws::Utils::Logging::LogLevel::Trace;
Aws::InitAPI(options);{
```

The good news here is that the manager is not auto-generated, so a PR on it is fairly simple to review and accept, so still glad to accept one if you have a fix in mind.

**Comment by ambitslix** [comment]
No problem, I had enabled tracing but didn't get any additional output. I didn't investigate why... I think the general solution should be that an example ( ListAvailableVoices) should return either some kind of success or failure. Not returning anything from a request should never happen especially considering there's a REST call behind it. Moreover an empty result is ambiguous in this case. What threw me off was the empty result and no error. I interpreted this as an empty list and no voices being available. So my solution would be to return the list if the REST call is successful otherwise return the HTTP error response. It's common practice with examples to hand wave error checking etc which is mostly justified so I'm just suggesting to be verbose in the case that something went wrong. "Consider examples to be End-User-Products"

I'm not sure what you referring to with auto generated manager?
